src/Compilers/CSharp/Portable/BoundTree/DecisionTree.cs

```diff
@@ -0,0 +1,916 @@
+                        {
+                            // duplicate switch label will have been reported during initial binding.
+                        }
+                        else
+                        {
+                            defaultLabel = label;
+                        }
+                    }
+                    else
+                    {
+                        this._syntax = label.Syntax;
+                        var subsumedErrorCode = CheckSubsumed(label.Pattern, result, expressionIsNull);
+                        if (subsumedErrorCode != 0 && subsumedErrorCode != ErrorCode.ERR_NoImplicitConvCast)
+                        {
+                            if (!label.HasErrors)
+                            {
+                                _diagnostics.Add(subsumedErrorCode, label.Pattern.Syntax.Location);
+                            }
+                        }
+                        else
+                        {
+                            AddToDecisionTree(result, label.Pattern, label.Guard, label);
+                        }
+                    }
+                }
+            }
+
+            if (defaultLabel != null)
+            {
+                Add(result, (e, t) => new DecisionTree.Guarded(_switchStatement.Expression, _switchStatement.Expression.Type, default(ImmutableArray<KeyValuePair<BoundExpression, LocalSymbol>>), null, defaultLabel));
+            }
+
+            return result;
+        }
+
+        private void AddToDecisionTree(DecisionTree decisionTree, BoundPattern pattern, BoundExpression guard, BoundPatternSwitchLabel label)
+        {
+            if (guard?.ConstantValue == ConstantValue.False) return;
+            switch (pattern.Kind)
+            {
+                case BoundKind.ConstantPattern:
+                    {
+                        var constantPattern = (BoundConstantPattern)pattern;
+                        AddByValue(decisionTree, constantPattern.Value, (e, t) => new DecisionTree.Guarded(e, t, default(ImmutableArray<KeyValuePair<BoundExpression, LocalSymbol>>), guard, label), label.HasErrors);
+                        break;
+                    }
+                case BoundKind.DeclarationPattern:
+                    {
+                        var declarationPattern = (BoundDeclarationPattern)pattern;
+                        DecisionMaker maker = (e, t) => new DecisionTree.Guarded(e, t, ImmutableArray.Create(new KeyValuePair<BoundExpression, LocalSymbol>(e, declarationPattern.LocalSymbol)), guard, label);
+                        if (declarationPattern.IsVar)
+                        {
+                            Add(decisionTree, maker);
+                        }
+                        else
+                        {
+                            AddByType(decisionTree, declarationPattern.DeclaredType.Type, maker);
+                        }
+                        break;
+                    }
+                case BoundKind.WildcardPattern:
```

Comment:
That would make the comment difficult. I'll add a default to this case block.

---
In reply to: [64672482](https://github.com/dotnet/roslyn/pull/11485#discussion_r64672482) [](ancestors = 64672482)